process.c

```diff
@@ -1274,48 +1279,135 @@ proc_wait(int argc, VALUE *argv)
+ *
+ *      pid0 = Process.spawn('ruby', '-e', 'exit 13') # => 230866
+ *      pid1 = Process.spawn('ruby', '-e', 'exit 14') # => 230891
+ *      Process.wait(pid0)                            # => 230866
+ *      $?                                            # => #<Process::Status: pid 230866 exit 13>
+ *      Process.wait(pid1)                            # => 230891
+ *      $?                                            # => #<Process::Status: pid 230891 exit 14>
+ *      Process.wait(pid0)                            # Raises Errno::ECHILD
+ *
+ *  - <tt>0</tt>: Waits for any child process whose group ID
+ *    is the same as that of the current process:
+ *
+ *      parent_pgpid = Process.getpgid(Process.pid)
+ *      puts "Parent process group ID is #{parent_pgpid}."
+ *      child0_pid = fork do
+ *      puts "Child 0 pid is #{Process.pid}"
+ *      child0_pgid = Process.getpgid(Process.pid)
+ *        puts "Child 0 process group ID is #{child0_pgid} (same as parent's)."
+ *      end
+ *      child1_pid = fork do
+ *        puts "Child 1 pid is #{Process.pid}"
+ *        Process.setpgid(0, Process.pid)
+ *        child1_pgid = Process.getpgid(Process.pid)
+ *        puts "Child 1 process group ID is #{child1_pgid} (different from parent's)."
+ *      end
+ *      retrieved_pid = Process.wait(0)
+ *      puts "Process.wait(0) returned pid #{retrieved_pid}, which is child 0 pid."
+ *      begin
+ *        Process.wait(0)
+ *      rescue Errno::ECHILD => x
+ *        puts "Raised #{x.class}, because child 1 process group ID differs from parent process group ID."
+ *      end
+ *
+ *    Output:
+ *
+ *      Parent process group ID is 225764.
+ *      Child 0 pid is 225788
+ *      Child 0 process group ID is 225764 (same as parent's).
+ *      Child 1 pid is 225789
+ *      Child 1 process group ID is 225789 (different from parent's).
+ *      Process.wait(0) returned pid 225788, which is child 0 pid.
+ *      Raised Errno::ECHILD, because child 1 process group ID differs from parent process group ID.
+ *
+ *  - <tt>-1</tt> (default): Waits for any child process:
+ *
+ *      parent_pgpid = Process.getpgid(Process.pid)
+ *      puts "Parent process group ID is #{parent_pgpid}."
+ *      child0_pid = fork do
+ *        puts "Child 0 pid is #{Process.pid}"
+ *        child0_pgid = Process.getpgid(Process.pid)
+ *        puts "Child 0 process group ID is #{child0_pgid} (same as parent's)."
+ *      end
+ *      child1_pid = fork do
+ *        puts "Child 1 pid is #{Process.pid}"
+ *        Process.setpgid(0, Process.pid)
+ *        child1_pgid = Process.getpgid(Process.pid)
+ *        puts "Child 1 process group ID is #{child1_pgid} (different from parent's)."
+ *      end
+ *      retrieved_pid = Process.wait(-1)
+ *      puts "Process.wait(-1) returned pid #{retrieved_pid}, which is child 0 pid."
+ *      retrieved_pid = Process.wait(-1)
+ *      puts "Process.wait(-1) returned pid #{retrieved_pid}, which is child 1 pid."
```

Comment:
I don't think we can assume which of the `retrieved_pid` is child 0 and child 1. The ordering will depend on which one of the two children finish first.

Reply:
Fixed (child 1 sleeps).